packages/oracle-node/src/fetchers/binance/BinanceClient.ts

```diff
@@ -0,0 +1,134 @@
+import { v4 as uuidv4 } from "uuid";
+import { WebSocketClient, WebSocketMessage } from "../WebSocketClient";
+import { config } from "../../config";
+
+export interface BinanceRequest {
+  id: string;
+  method: string;
+  params: {
+    symbols: string[];
+  };
+}
+
+interface BinanceResponse {
+  id: string;
+  status: number;
+  result: unknown;
+}
+
+export interface BinanceTickersResponse extends BinanceResponse {
+  result: {
+    symbol: string;
+    price: number;
+  }[];
+}
+
+type PendingRequests = {
+  [key: string]: {
+    onSuccess: (response: BinanceResponse) => void;
+    onError: (error: Error) => void;
+    timeout: NodeJS.Timeout;
+  };
+};
+
+const REQUEST_TIMEOUT_MS = 5000;
+
+export class BinanceClient {
+  private client: WebSocketClient = new WebSocketClient(
+    config.binanceWebsocketApiUrl,
+    {
+      onMessage: this.onMessage.bind(this),
+      onClose: this.onClose.bind(this),
+      onError: this.onError.bind(this),
+    },
+    {
+      maxConnectionTimeMs: 1000 * 60 * 10,
```

Comment:
Binance API allows to keep connections for max 24h. I used 10h here because it doesn't really matter performance-wise, so let's refresh more often.